### PR TITLE
Full support for HTTP methods

### PR DESCRIPTION
**Pull request by incubos**
Implemented client and server support for the following HTTP methods:
- `PUT`
- `DELETE`
- `TRACE`
- `CONNECT`
- `PATCH`

Added unit tests.

### CODE REVIEW

File: src/one/nio/http/HttpSession.java [hunk 1]
```diff
@@ -137,12 +137,22 @@ protected Request parseRequest(byte[] buffer, int start, int length) throws Http
...
+        } else if (length > 15 && Utf8.startsWith(Request.VERB_TRACE, buffer, start)) {
+            return new Request(Request.METHOD_TRACE, Utf8.read(buffer, start + 6, length - 15), http11);
+        } else if (length > 15 && Utf8.startsWith(Request.VERB_PATCH, buffer, start)) {
+            return new Request(Request.METHOD_PATCH, Utf8.read(buffer, start + 6, length - 15), http11);
+        } else if (length > 16 && Utf8.startsWith(Request.VERB_DELETE, buffer, start)) {
+            return new Request(Request.METHOD_DELETE, Utf8.read(buffer, start + 7, length - 16), http11);
         } else if (length > 17 && Utf8.startsWith(Request.VERB_OPTIONS, buffer, start)) {
             return new Request(Request.METHOD_OPTIONS, Utf8.read(buffer, start + 8, length - 17), http11);
+        } else if (length > 17 && Utf8.startsWith(Request.VERB_CONNECT, buffer, start)) {
+            return new Request(Request.METHOD_CONNECT, Utf8.read(buffer, start + 8, length - 17), http11);
```
Comment (apangin) [inline src/one/nio/http/HttpSession.java]: Looks like it's time for refactoring

Comment (incubos) [reply src/one/nio/http/HttpSession.java]: I will refactor it.

File: src/one/nio/http/HttpSession.java [hunk 2]
```diff
@@ -134,15 +134,22 @@ protected int processHttpBuffer(byte[] buffer, int length) throws IOException, H
...
+            final byte[] verb = Request.VERBS[i];
+            if (length > verb.length && Utf8.startsWith(verb, buffer, start)) {
+                return new Request(
+                        i,
+                        Utf8.read(
+                                buffer,
+                                // Strip "<VERB> "
+                                start + verb.length,
+                                // Trim " HTTP/1.{0,1}"
+                                length - (verb.length + 9)),
```
Comment (apangin) [inline src/one/nio/http/HttpSession.java]: Won't it be ArrayIndexOutBoundsException for a malformed request like "GET \n"

Comment (incubos) [reply src/one/nio/http/HttpSession.java]: Yes, you are right. Added a unit test and a fix.

**Pull request merged**